desktop/cmp/persistenceManager/PersistenceManager.ts

```diff
@@ -0,0 +1,214 @@
+import {div, fragment, hbox} from '@xh/hoist/cmp/layout';
+import {hoistCmp, HoistProps, PlainObject, uses} from '@xh/hoist/core';
+import {button} from '@xh/hoist/desktop/cmp/button';
+import {Icon} from '@xh/hoist/icon/Icon';
+import {menu, menuDivider, menuItem, popover} from '@xh/hoist/kit/blueprint';
+import {groupBy, keys, sortBy} from 'lodash';
+import {ReactNode} from 'react';
+import {manageDialog} from './impl/ManageDialog';
+import {saveDialog} from './impl/SaveDialog';
+import './PersistenceManager.scss';
+import {PersistenceManagerModel} from './PersistenceManagerModel';
+
+interface PersistenceManagerModelProps extends HoistProps {
+    model?: PersistenceManagerModel;
+    /** True to disable options for saving/managing items. */
+    minimal?: boolean;
+}
+
+export const [PersistenceManager, persistenceManager] =
+    hoistCmp.withFactory<PersistenceManagerModelProps>({
+        displayName: 'PersistenceManager',
+        model: uses(PersistenceManagerModel),
+
+        render({model, minimal = false}) {
+            const {
+                selectedObject,
+                isShared,
+                capitalPluralNoun,
+                manageDialogModel,
+                saveDialogModel
+            } = model;
+
+            return fragment(
+                hbox({
+                    className: 'persistence-manager',
+                    items: [
+                        popover({
+                            item: button({
+                                text: selectedObject?.name
+                                    ? getHierarchyDisplayName(selectedObject.name)
+                                    : capitalPluralNoun,
+                                icon: isShared ? Icon.users() : Icon.bookmark(),
+                                rightIcon: Icon.chevronDown(),
+                                outlined: true
+                            }),
+                            content: div({
+                                items: [
+                                    div({className: 'xh-popup__title', item: capitalPluralNoun}),
+                                    objMenu({minimal})
+                                ]
+                            }),
+                            placement: 'bottom-start'
+                        }),
+                        saveButton()
+                    ]
+                }),
+                manageDialogModel ? manageDialog({key: manageDialogModel.xhId}) : null,
```

Comment:
I don't think we need to specify the `key` to react to model changes anymore. Might be more Hoisty to use `omit` here rather than the ternary